extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/runtime/OidcCommonUtils.java

```diff
@@ -504,8 +505,10 @@ public static Map<OidcEndpoint.Type, List<OidcRequestFilter>> getOidcRequestFilt
             for (OidcRequestFilter filter : container.listAll(OidcRequestFilter.class).stream().map(handle -> handle.get())
                     .collect(Collectors.toList())) {
                 OidcEndpoint endpoint = ClientProxy.unwrap(filter).getClass().getAnnotation(OidcEndpoint.class);
-                OidcEndpoint.Type type = endpoint != null ? endpoint.value() : OidcEndpoint.Type.ALL;
-                map.computeIfAbsent(type, k -> new ArrayList<OidcRequestFilter>()).add(filter);
+                OidcEndpoint.Type[] types = endpoint != null ? endpoint.value() : ALL_TYPES;
+                for (OidcEndpoint.Type type : types) {
+                    map.computeIfAbsent(type, k -> new ArrayList<OidcRequestFilter>()).add(filter);
+                }
```

Comment:
That works too, but I was thinking of avoiding that array with a single if/else: 

```suggestion
				if (endpoint != null) {
	                for (OidcEndpoint.Type type : endpoint.value()) {
	                    map.computeIfAbsent(type, k -> new ArrayList<OidcRequestFilter>()).add(filter);
                } else {             
		           map.computeIfAbsent(OidcEndpoint.Type.ALL, k -> new ArrayList<OidcRequestFilter>()).add(filter);
                }
```

Reply:
@gastaldi Sure